eq-author/src/App/history/HistoryItem.js

```diff
@@ -37,27 +75,179 @@ const EventText = styled.div`
+    margin-right: -1.5em !important;
+  }
+`;
+
+const SaveButton = styled(Button)`
+  font-size: 0.75em;
+  margin-right: -1em;
+  margin-bottom: 1em;
+`;
+
+const EditButtonIcon = styled.button`
+  display: inline-flex;
+  flex: 0 0 auto;
+  color: ${colors.grey};
+  padding: 1em;
+  border-radius: ${radius};
+  font-size: 1em;
+  font-weight: 600;
+  cursor: pointer;
+  line-height: 1;
+  justify-content: center;
+  -webkit-font-smoothing: antialiased;
+  text-rendering: optimizeLegibility;
+  text-decoration: none;
+  transition: all 100ms ease-out;
+  letter-spacing: 0;
+  position: relative;
+  overflow: hidden;
+  border: 1px solid var(--color-bg);
+  background: url(${IconDelete}) no-repeat center center;
+  height: 15px;
+  width: 15px;
+  &:focus,
+  &:active {
+    outline-width: 0;
+  }
+  &:focus {
+    box-shadow: 0 0 0 3px ${colors.tertiary};
+    outline: none;
+  }
+  &[disabled] {
+    pointer-events: none;
+    opacity: 0.6;
+  }
+`;
+
 const formatDate = unformattedDate =>
   moment(unformattedDate).format("DD/MM/YYYY [at] HH:mm");
 
+const RTEWrapper = styled.div`
+  margin: 0 2em 0 1em;
+  padding-bottom: 1em;
+  width: 100%;
+`;
+
 const HistoryItem = ({
   questionnaireTitle,
+  questionnaireId,
+  itemId,
   publishStatus,
   userName,
+  currentUser,
+  userId,
   bodyText,
+  type,
   createdAt,
 }) => {
+  const [activeEdit, setActiveEdit] = useState(false);
+  const [noteState, setNoteState] = useState(bodyText);
+
+  const [updateNote] = useMutation(UPDATE_NOTE);
+  const [deleteNote] = useMutation(DELETE_NOTE, {
+    update(
+      cache,
+      {
+        data: { deleteHistoryNote },
+      }
+    ) {
+      cache.writeQuery({
+        query: questionnaireHistoryQuery,
+        variables: {
+          input: { questionnaireId },
+        },
+        data: { history: deleteHistoryNote },
+      });
+    },
+  });
+
+  const handleDelete = () => {
+    deleteNote({
+      variables: {
+        input: {
+          id: itemId,
+          questionnaireId,
+        },
+      },
+    });
+  };
+
+  const handleEdit = () => {
+    setActiveEdit(true);
+  };
+
+  const handleSave = () => {
+    setActiveEdit(false);
+
+    updateNote({
+      variables: {
+        input: {
+          id: itemId,
+          questionnaireId,
+          bodyText: noteState,
+        },
+      },
+    });
+  };
+
   return (
-    <StyledItem>
-      <QuestionnaireTitle>
-        {questionnaireTitle} - <strong>{publishStatus}</strong>
-      </QuestionnaireTitle>
-      <QuestionnaireUserName>
-        {userName} - {formatDate(createdAt)}
-      </QuestionnaireUserName>
-      <BreakLine />
-      {bodyText && <EventText dangerouslySetInnerHTML={{ __html: bodyText }} />}
-    </StyledItem>
+    <EditableItem>
+      {(currentUser.id === userId || currentUser.admin) &&
+        (type === "note" && (
+          <EditThis>
+            <HistoryButtonGroup horizontal align="right">
+              <EditButtonIcon onClick={handleEdit} />
```

Comment:
```suggestion
              <EditButtonIcon onClick={() => setActiveEdit(true))} />
```
And you can remove the function.